internal/webhooks/iam/v1alpha1/userdeactivation_webhook.go

```diff
@@ -54,6 +57,23 @@ func (m *UserDeactivationMutator) Default(ctx context.Context, obj runtime.Objec
 
 	userdeactivationlog.Info("Defaulting deactivatedBy complete", "name", ud.GetName(), "deactivatedBy", ud.Spec.DeactivatedBy)
 
+	// Set the owner reference so the UserDeactivation is garbage collected when the User is deleted.
+	// The user is guaranteed to exist, as ValidateCreate validates that the user exists
+	user := &iamv1alpha1.User{}
+	if err := m.client.Get(ctx, client.ObjectKey{Name: ud.Spec.UserRef.Name}, user); err != nil {
+		userdeactivationlog.Error(err, "failed to fetch referenced User while setting owner reference", "userName", ud.Spec.UserRef.Name)
+		return errors.NewInternalError(fmt.Errorf("failed to fetch referenced User while setting owner reference, %w", err))
+	}
+
+	ud.OwnerReferences = []metav1.OwnerReference{
+		{
+			APIVersion: iamv1alpha1.SchemeGroupVersion.String(),
+			Kind:       "User",
+			Name:       user.Name,
+			UID:        user.GetUID(),
+		},
+	}
```

Comment:
You can simplify this by using the `controllerutil.SetOwnerReference()` function. 

```suggestion
if err := SetOwnerReference(user, ud, m.scheme); err != nil {
     return errors.NewInternalError(fmt.Errorf("failed to set owner reference for user deactivation: %w", errr))
}
```

You'll want to add a new `scheme` field to the webhook and use `mgr.GetScheme()` to set the scheme in the `SetupUserDeactivationWebhooksWithManager()` function.